src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -369,6 +382,11 @@ internal override void Validate()
             {
                 throw new InvalidOperationException(MsalErrorMessage.InvalidRedirectUriReceived(Config.RedirectUri));
             }
+
+            if (Config.IsBrokerEnabled && Config.MultiCloudSupport)
+            {

```

Comment:
You may want to do this check prior to assigning COnfig.RedirectUri

Reply:
I have added a check to throw NotSupportedException if MultiCloudSupport is opted with Broker. I can remove the broker check from here. Is this what you are asking for @SameerK-MSFT ?